assets/js/modules/tagmanager/hooks/useExistingTagEffect.js

```diff
@@ -33,14 +34,29 @@ export default function useExistingTagEffect() {
 	const existingTag = useSelect( ( select ) => select( STORE_NAME ).getExistingTag() );
 	const existingTagPermission = useSelect( ( select ) => select( STORE_NAME ).getTagPermission( existingTag ) );
 	const hasExistingTagPermission = useSelect( ( select ) => select( STORE_NAME ).hasExistingTagPermission() );
-	// Set the accountID and containerID if there is an existing tag.
-	const { selectAccount, selectContainerByID } = useDispatch( STORE_NAME );
+
+	const singleAnalyticsPropertyID = useSelect( ( select ) => select( STORE_NAME ).getSingleAnalyticsPropertyID() );
+	const analyticsModuleActive = useSelect( ( select ) => select( CORE_MODULES ).isModuleActive( 'analytics' ) );
+
+	const { selectAccount, selectContainerByID, setGAPropertyID } = useDispatch( STORE_NAME );
+
 	useEffect( () => {
 		( async () => {
 			if ( hasExistingTag && hasExistingTagPermission ) {
 				await selectAccount( existingTagPermission.accountID );
 				await selectContainerByID( existingTag );
 			}
+
+			// If a singular property ID is set in the container(s) and Analytics is active,
+			// we store the property ID which prevents the Analytics module from including
+			// the snippet using the backend googlesitekit_analytics_can_use_snippet hook to
+			// prevent duplicate measurement.
+			// The stored property ID is then used to re-enable the Analytics snippet if
+			// Tag Manager is disconnected in future.
+			if ( singleAnalyticsPropertyID !== undefined && analyticsModuleActive ) {
+				// Set the GA property ID in the Tag Manager store.
+				await setGAPropertyID( singleAnalyticsPropertyID === null ? '' : singleAnalyticsPropertyID );
+			}
 		} )();
-	}, [ hasExistingTag, existingTag, hasExistingTagPermission, existingTagPermission ] );
+	}, [ hasExistingTag, existingTag, hasExistingTagPermission, existingTagPermission, singleAnalyticsPropertyID, analyticsModuleActive ] );
```

Comment:
I don't think the effect needs to change here since it isn't related to existing tags. In GA, we do however treat a GA tag in a GTM container as an existing tag, but this is a bit different.

We essentially need a `useGAPropertyIDEffect` that synchronizes the current value based on the selected containers.
This is fairly straightforward since the `singleAnalyticsPropertyID` is already based on container selections.

```js
function useGAPropertyIDEffect() {
	const singleAnalyticsPropertyID = useSelect( ( select ) => select( STORE_NAME ).getSingleAnalyticsPropertyID() );
	const { setGAPropertyID } = useDispatch( STORE_NAME );

	useEffect( () => {
		if ( singleAnalyticsPropertyID !== undefined ) {
			setGAPropertyID( singleAnalyticsPropertyID || '' );
		}
	}, [ singleAnalyticsPropertyID ] );
}
```
This would then be called in the same places where Tag Manager's `useExistingTagEffect` is called (setup + settings). That way, the setting is kept in sync with the current `singleAnalyticsPropertyID` and will be saved automatically when settings are saved.